elftools/dwarf/ranges.py

```diff
@@ -114,14 +114,21 @@ def get_range_list_at_offset_ex(self, offset):
         return struct_parse(self.structs.Dwarf_rnglists_entries, self.stream, offset)
 
     def iter_range_lists(self):
-        """ Yield all range lists found in the section according to readelf rules.
+        """ Yields all range lists found in the section according to readelf rules.
         Scans the DIEs for rangelist offsets, then pulls those.
+        Returned rangelists are always translated into lists of BaseAddressEntry/RangeEntry objects.
         """
-        # Calling parse until the stream ends is wrong, because ranges can overlap.
-        # Need to scan the DIEs to know all range locations
+        # Rangelists can overlap. That is, one DIE points at the rangelist beginning, and another
+        # points at the middle of the same. Therefore, enumerating them is not a well defined
+        # operation - do you count those as two different (but overlapping) ones, or as a single one?
+        # For debugging utility, you want two. That's what readelf does. For faithfully
+        # representing the section contents, you want one.
+        # That was the behaviour of pyelftools 0.28 and below - calling
```

Comment:
Not sure I get this part -- the behavior doesn't appear to change in this PR, so what is the meaning of "this was the behavior of ... 0.28 and below"? Is something changing?

The docstring of this method already says "according to readelf rules", so it's clear which way the choice is made

Reply:
The long comment is me ~venting~ explaining why "list all rangelists" is a not well defined operation. I changed the pyelftools' behavior on that long time ago, back when implementing DWARFv5.